patches/extra/ungoogled-chromium/add-flag-for-disabling-link-drag.patch

```diff
@@ -0,0 +1,67 @@
+--- a/third_party/blink/common/features.cc
++++ b/third_party/blink/common/features.cc
+@@ -14,6 +14,8 @@
+ 
+ namespace blink {
+ namespace features {
++
++const base::Feature kDisableLinkDrag{"DisableLinkDrag", base::FEATURE_DISABLED_BY_DEFAULT};
+ 
+ // Apply lazy-loading to ad frames which have embeds likely impacting Core Web
+ // Vitals.
+--- a/third_party/blink/public/common/features.h
++++ b/third_party/blink/public/common/features.h
+@@ -16,6 +16,8 @@
+ 
+ namespace blink {
+ namespace features {
++
++BLINK_COMMON_EXPORT extern const base::Feature kDisableLinkDrag;
+ 
+ BLINK_COMMON_EXPORT extern const base::Feature kAutomaticLazyFrameLoadingToAds;
+ BLINK_COMMON_EXPORT extern const base::Feature
+--- a/third_party/blink/renderer/core/input/mouse_event_manager.cc
++++ b/third_party/blink/renderer/core/input/mouse_event_manager.cc
+@@ -687,8 +687,14 @@
+ 
+   bool single_click = event.Event().click_count <= 1;
+ 
++if (base::FeatureList::IsEnabled(features::kDisableLinkDrag)){
++  mouse_down_may_start_drag_ = single_click && !IsSelectionOverLink(event) &&
++                               !IsExtendingSelection(event) &&
++			        !event.GetHitTestResult().IsSelected(event.GetHitTestLocation());
++}else{
+   mouse_down_may_start_drag_ = single_click && !IsSelectionOverLink(event) &&
+                                !IsExtendingSelection(event);
++}
+ 
+   mouse_down_ = event.Event();
+ 
+--- a/third_party/blink/renderer/core/editing/selection_controller.cc
++++ b/third_party/blink/renderer/core/editing/selection_controller.cc
+@@ -1319,9 +1319,13 @@
+ }
+ 
+ bool IsSelectionOverLink(const MouseEventWithHitTestResults& event) {
++if (base::FeatureList::IsEnabled(features::kDisableLinkDrag)){
++  return event.IsOverLink();
++}else{
+   return (event.Event().GetModifiers() & WebInputEvent::Modifiers::kAltKey) !=
+              0 &&
+          event.IsOverLink();
++}
+ }
+ 
+ bool IsUserNodeDraggable(const MouseEventWithHitTestResults& event) {
+--- a/chrome/browser/ungoogled_flag_entries.h
++++ b/chrome/browser/ungoogled_flag_entries.h
+@@ -108,4 +108,8 @@
+      "Hide SidePanel Button",
+      "Hides the SidePanel Button.  ungoogled-chromium flag.",
+      kOsDesktop, SINGLE_VALUE_TYPE("hide-sidepanel-button")},
++    {"disable-link-drag",
++     "Disable link drag",
++     "Prevents dragging of links and selected text.  ungoogled-chromium flag.",
```

Comment:
I know its everywhere but is there an actual reason to have two spaces before "ungoogled-chromium flag."?

Reply:
Hmm, good catch! I didn't even notice that! Perfectionist in me would be definitely willing to replace all double-spaces unless there is a reason for them to be there.

Reply:
>Everywhere: Fix formatting issue

*breaks production*

Reply:
I guess, it would be safe to remove all double-spaces since they are not rendered anyways:
![image](https://user-images.githubusercontent.com/10319700/188684962-2192f881-a849-4b94-9f65-3d8bb5364f8a.png)

Reply:
The double space thing comes from typewriters. Some people/organisations still recommend writing that way. There're many articles on this matter if you do a search for "double space after period" (e.g. https://smithsonianmag.com/smart-news/microsoft-word-now-considers-two-spaces-after-period-error-180974754).

PS You've missed a few double spaces in between the actual sentences, so you may want to have another look if you do intend to replace all of them 😉.

Reply:
Indeed some of the double-spaces come from Chromium source itself. For example [here](https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/prefetch/prefetch_prefs.h;l=31-38?q=%22%2F%2F+Enum+representing+possible+values+of+the+Preload+Pages+opt-in+state.++Since%22&ss=chromium). Still it is funny, how inconsistent it is: there are two spaces after the first sentence, but single spaces further down along this comment.

> PS You've missed a few double spaces in between the actual sentences, so you may want to have another look if you do intend to replace all of them .

Behold my wrath against multiple-spaces! :D Honestly saying I hate repetitive spaces because that reminds me of the times I had to work with documents whose formatting was done solely by spaces! Wild hatred!
BTW, @tomasz1986 it would be great if you could test this PR.

Reply:
> Still it is funny, how inconsistent it is: there are two spaces after the first sentence, but single spaces further down along this comment.

It could be due to different people making changes to the text 😉.

> BTW, @tomasz1986 it would be great if you could test this PR.

I'd love to, but I don't really compile the browser myself 🙁. I've given it a try multiple times, but my hardware isn't really sufficient for it yet.

Reply:
I was taught an oldschool method of typing growing up so I still have have a habit of adding double-spaces after ending sentences.  I probably added that when doing a flag and it got copy/pasted it over to others over time.  I guess those could be considered age markers 😛.

Reply:
So in the end I hope no one would be against me removing them, right?

Reply:
are you satoshi nakamoto?

Reply:
> So in the end I hope no one would be against me removing them, right?

That should be fine!